augur/io/tabular_file.py

```diff
@@ -0,0 +1,76 @@
+import csv
+import pandas as pd
+from typing import List, Optional
+
+from augur.errors import AugurError
+
+from .file import open_file
+
+
+class TabularFile:
+    def __init__(self, file:str,
+            header:Optional[bool]=True,
+            names:Optional[List[str]]=None):
+        self.file = file
+        """File to be loaded."""
+
+        self.header = header
+        """Indicates whether a header is present in the file."""
+
+        self.names = names or []
+        """Column names if a header is not present."""
```

Comment:
These details are already exposed through the API autodocs: https://nextstrain--1094.org.readthedocs.build/projects/augur/en/1094/api/developer/augur.io.tabular_file.html

However, I can switch to numpy-style since that is more prevalent in this codebase.

I do think that typing is better defined inline rather than in docstrings. Maybe there's a way to expose Python type annotations through the API autodocs.

Reply:
I believe for autodoc to pick up the type annotations on the instance attributes, the attributes need to be explicitly annotated, e.g. `self.file: str = file`.

While type checkers will do type inference and know that the type of `file` is `str` so that's type of `self.file`, I don't think autodoc does this.